src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -176,61 +207,214 @@ def __blob_is_not_complete(metadata):
     return True
 
 
-def acr_build_queue(cmd,
+def acr_queue(cmd,
               client,
               registry_name,
-              source_location,
-              docker_file_path,
-              image_name,
+              image_name=None,
+              docker_file_path=None,
               resource_group_name=None,
-              context_path=None,
+              source_location=None,
               timeout=None,
-              arguments=None,
-              secret_arguments=None):
+              build_args=None,
+              secret_build_args=None,
+              show_logs='true'):
 
     resource_group_name = get_resource_group_name_by_registry_name(
         cmd.cli_ctx, registry_name, resource_group_name)
-    build_parameters = DockerBuildParameters(docker_file_path)
 
-    if context_path:
-        build_parameters.context_path = context_path
+    if docker_file_path is None:
+        docker_file_path = "Dockerfile"
+
+    build_parameters = DockerBuildParameters(docker_file_path)
+    # context_path in tar build is alwasy the source code root folder
+    build_parameters.context_path = "."
+
+    if source_location is None:
+        source_location = "."
+
+    if os.path.exists(source_location):
+        if os.path.isdir(source_location):
+            source_location = _upload_source_code(
+                client, registry_name, resource_group_name, source_location)
+        else:
+            raise CLIError(
+                "'--source-location' should be a local directory path or remote url.")
+    else:
+        source_location = _check_remote_source_code(source_location)
 
-    # [Server Bug] we should allow skiping image_name if users don't want to push
     is_push_enabled = True
     if image_name is None:
         is_push_enabled = False
+        print("'--tag' is not providied. Skip image push after build.")
+    else:
+        image_name = _check_image_name(image_name)
 
     # hard-code platform to linux and cpu to 1
     platform = PlatformProperties("Linux")
     platform.cpu = 1
```

Comment:
>cpu = 1 [](start = 13, length = 7)

This should default to 2 core.

Reply:
We currently default to 1. Why 2?

Reply:
Shouldn't this be a swagger level default value? Add to swagger if it's not there and don't set it in CLI.

Reply:
I believe you allocate 2 cores by default. /cc @mnltejaswini do we validate this on the RP?

Reply:
We will update it once we settle down the build configuration.